src/enclave/rpc_sessions.h

```diff
@@ -75,26 +114,41 @@ namespace enclave
         throw std::logic_error(
           "Duplicate conn ID received inside enclave: " + std::to_string(id));
 
-      if (sessions.size() >= max_open_sessions)
+      if (sessions.size() >= max_open_sessions_hard)
       {
         LOG_INFO_FMT(
           "Refusing a session inside the enclave - already have {} sessions "
           "and limit is {}: {}",
           sessions.size(),
-          max_open_sessions,
+          max_open_sessions_hard,
           id);
 
         RINGBUFFER_WRITE_MESSAGE(
           tls::tls_stop, to_host, id, std::string("Session refused"));
-        return;
       }
+      else if (sessions.size() >= max_open_sessions_soft)
+      {
+        LOG_INFO_FMT(
```

Comment:
This seems rather verbose?

Reply:
I have this at `INFO` to match the existing logging when the cap is reached. I think it shouldn't happen under normal operation, and whenever it _does_ happen the operator should be informed. Especially with the other change from #2521, where the operator is likely to be using a separate interface and not seeing this cap, its important that they have some signal that the service is overloaded. Maybe with #2522 they have another way to find this, but I think the log is still valuable.